node/src/utils/external.rs

```diff
@@ -40,9 +40,11 @@ pub static RESOURCES_PATH: Lazy<PathBuf> =
 /// An `External` also always provides a default, which will always result in an error when `load`
 /// is called. Should the underlying type `T` implement `Default`, the `with_default` can be
 /// used instead.
+// TODO: `Loaded` is only ever used in tests, refactor this
+#[allow(unused)]
 #[derive(Clone, DataSize, Eq, Debug, Deserialize, PartialEq, Serialize)]
 #[serde(untagged)]
-pub enum External<T> {
+pub(crate) enum External<T> {
     /// Value that should be loaded from an external path.
     Path(PathBuf),
     /// Loaded or immediate value.
```

Comment:
`Loaded` is only used in test, we should get rid of it.  CC @marc-casperlabs

Reply:
If we're not doing `Loaded` at all, the whole type should go away. The sole purpose of the enum was to make it easy to supply custom values for configuration that may be in an external file in a test. Temporary files have to be used instead if this is removed. I'm not opposed to that.

So maybe we want to go a step further and remove the logic from the configuration entirely. Components would then themselves be responsible for loading data from a file.

This does move the point of failure back a bit (from configuration load to execution), but maybe the reduced complexity is worth it.

Reply:
As discussed, we should consider iterating on this later.

Reply:
Removed spurious comment in 7f003c0b3a8ac58d47082a3a36ed50ab85d5c5d8